app/models/automation_worker.rb

```diff
@@ -15,5 +15,6 @@ def configure_worker_deployment(definition, replicas = 0)
     super
 
     definition[:spec][:template][:spec][:serviceAccountName] = "manageiq-automation"
+    definition[:spec][:template][:spec][:containers][0][:env]["AUTOMATION_JOB_SERVICE_ACCOUNT"] = ENV.fetch("WORKER_SERVICE_ACCOUNT")
```

Comment:
As discussed, we will change this in the future to use a dedicated account named something like `manageiq-restricted`, but this is good for now since manageiq-default is already restricted.